providers/gce/gce_instances.go

```diff
@@ -92,6 +93,34 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 	return refs
 }
 
+// orderAddresses orders node IP addresses:
+//   - In single-stack IPv6 clusters IPv6 addresses before the IPv4 addresses.
+//   - In other clusters IPv4 addresses before IPv6 addresses.
+func (g *Cloud) orderAddresses(addresses []v1.NodeAddress) []v1.NodeAddress {
```

Comment:
Please make this a free function in a follow on commit.

```
func orderAddresses(stackType, addresses) ...
```

It will reduce the amount of dependencies for unit tests.